### PR TITLE
feat(input-opt): add input OTP masking options on parent, group and slot

### PR DESCRIPTION
**Pull request by liam-langstaff**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our
      guidelines: https://github.com/spartan-ng/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### Primitives

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] autocomplete
- [ ] avatar
- [ ] badge
- [ ] breadcrumb
- [ ] button
- [ ] button-group
- [ ] calendar
- [ ] card
- [ ] carousel
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] empty
- [ ] dropdown-menu
- [ ] field
- [ ] form-field
- [ ] hover-card
- [ ] icon
- [ ] input
- [ ] input-group
- [x] input-otp
- [ ] item
- [ ] kbd
- [ ] label
- [ ] menubar
- [ ] native-select
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] resizable
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] sidebar
- [ ] skeleton
- [ ] slider
- [ ] sonner
- [ ] spinner
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toggle
- [ ] toggle-group
- [ ] tooltip
- [ ] typography

### Others

- [ ] trpc
- [ ] nx
- [ ] repo
- [ ] cli

## What is the current behavior?

--

Adds configurable masking for OTP slots across Brain and Helm.
• Introduces OTP character masking with brief reveal-then-mask behavior.
• Adds mask-value pipe and tests to cover masking timing and edge cases.
• Adds BrnInputOtpMask directive for group-level masking control.
• Wires group masking into Helm via hlmInputOtpGroup host directive.
• Supports per-slot [mask] overrides in hlm-input-otp-slot.
• Defines mask precedence as: slot mask -> group mask -> root brn-input-otp mask.
• Keeps existing behavior unchanged when masking is not enabled.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

https://github.com/user-attachments/assets/666105e2-9fdb-4dbe-b471-4011461804a6

### CODE REVIEW

File: libs/brain/input-otp/src/lib/brn-input-otp-slot.ts [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 	selector: 'brn-input-otp-slot',
+	imports: [MaskValuePipe],
 	changeDetection: ChangeDetectionStrategy.OnPush,
 	host: {
 		'[attr.data-active]': '_slot().isActive',
 	},
 	template: `
-		{{ _slot().char }}
-
+		{{ _slot().char | maskValue: _hasMask() : 200 }}
```
Comment (liam-langstaff) [inline libs/brain/input-otp/src/lib/brn-input-otp-slot.ts:15]: Could make the delay dynamic

File: libs/brain/input-otp/src/lib/brn-input-otp-slot.ts [hunk 2]
```diff
@@ -22,6 +24,14 @@ export class BrnInputOtpSlot {
...
 	public readonly index = input.required<number, NumberInput>({ transform: numberAttribute });
+	public readonly mask = input<boolean | undefined>(undefined);
+
+	/** Optional group-level mask directive (provided by hlmInputOtpGroup hostDirectives) */
+	protected readonly _groupMask = inject(BrnInputOtpMask, { optional: true });
 
 	protected readonly _slot = computed(() => this._inputOtp.context()[this.index()]);
+
+	protected readonly _hasMask = computed(() => {
+		return this.mask() ?? this._groupMask?.mask() ?? this._inputOtp.mask();
```
Comment (liam-langstaff) [inline libs/brain/input-otp/src/lib/brn-input-otp-slot.ts:35]: This is priority driven, slot > group > parent

File: libs/brain/input-otp/src/lib/brn-input-otp.spec.ts [hunk 3]
```diff
@@ -363,4 +366,34 @@ describe('BrnInputOtp', () => {
 			expect(screen.queryByTestId('slot-3')).not.toBeInTheDocument();
 		});
 	});
+	describe('mask behavior', () => {
```
Comment (liam-langstaff) [inline libs/brain/input-otp/src/lib/brn-input-otp.spec.ts:369]: Could add more? Validates against a parent mask, not a group or slot..

**Comment by m-risto** [comment]
@goetzrobin @MerlinMoos Any updates on this. Would be great if we would not need to overwrite the default spartan here.
